File: java/src/main/java/factory/creator/SuperVehicleFactory.java
package factory.creator;

import factory.product.SuperBus;
import factory.product.SuperTruck;
import factory.product.Vehicle;
import factory.product.VehicleType;

public class SuperVehicleFactory extends VehicleFactory {
    @Override
    protected Vehicle createVehicle(VehicleType vehicleType) {
        return switch (vehicleType) {
            case BUS -> new SuperBus();
            case TRUCK -> new SuperTruck();
        };
    }
}

File: kotlin/src/main/kotlin/factory/creator/SuperVehicleFactory.kt
package factory.creator

import factory.product.SuperBus
import factory.product.SuperTruck
import factory.product.Vehicle
import factory.product.VehicleType

class SuperVehicleFactory : VehicleFactory() {
    override fun createVehicle(vehicleType: VehicleType): Vehicle {
        return when (vehicleType) {
            VehicleType.BUS -> SuperBus()
            VehicleType.TRUCK -> SuperTruck()
        }
    }
}